servers/physics_2d/joints_2d_sw.h

```diff
@@ -60,6 +60,12 @@ class Joint2DSW : public Constraint2DSW {
 		bias = 0;
 		max_force = max_bias = 3.40282e+38;
 	};
+
+	virtual ~Joint2DSW() {
+		for (int i = 0; i < get_body_count(); i++) {
+			get_body_ptr()[i]->remove_constraint(this, i);
```

Comment:
It should probably check for the pointer to be valid for each body, since that's what `PinJoint2DSW` was doing. In general, it would be safer to do that for both 2D and 3D joints because in some cases one of the bodies can be null (I think this is currently supported only with 3D joints, but it might change in the future).